jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -205,6 +205,14 @@ static PluginConfigurationProcessor processCommonConfiguration(
       jibContainerBuilder.setCreationTime(Instant.now());
     }
 
+    // Adds all the extra files.
+    if (Files.exists(rawConfiguration.getExtraDirectory())) {
+      jibContainerBuilder.addLayer(
+          JavaContainerBuilderHelper.extraDirectoryLayerConfiguration(
+              rawConfiguration.getExtraDirectory(),
+              rawConfiguration.getExtraDirectoryPermissions()));
```

Comment:
What do you think of doing `TaskCommon.convertPermissionsMap()` here instead of `RawConfiguration` calling it? I have a desire to have `RawConfiguration` as a very think adapter that merely returns config values without any processing or validation.

Reply:
Makes sense.

Reply:
Ah wait, I remember why it's like this, it's because Gradle and Maven define the permissions map differently; Maven uses a `List<PermissionConfiguration>`, and Gradle uses a `Map<String, String>`. So it doesn't seem like we can just directly return these from `RawConfiguration` without any processing, since the actual raw type is different between the two plugins.

Reply:
Ah, okay. Since they have different types, I think this conversion is necessary to be able to unify them.